online-store/src/app/app.ts

```diff
@@ -0,0 +1,278 @@
+            let checkboxValue = el.childNodes[2].childNodes[0] as HTMLInputElement
+            checkboxValue.innerText = `${sum}`;
+        });
+    }
+    sumofChange(newArr);
+
+    function search() {
+        let searchInput: HTMLInputElement | null = document.querySelector('.header-input');
+        let searchArr;
+        if (searchInput){
+            searchInput.oninput = function (){
+                let valThis = this as HTMLInputElement;
+                let val = valThis.value.trim();
+                console.log (val);
+                searchArr = newArr.filter((el) => {
+                    if (el.country.toLowerCase().includes(val.toLowerCase()) || 
+                        el.brand.toLowerCase().includes(val.toLowerCase()) || 
+                        el.title.toLowerCase().includes(val.toLowerCase()) || 
+                        el.surface.toLowerCase().includes(val.toLowerCase()) || 
+                        el.drawing.toLowerCase().includes(val.toLowerCase())) {
+                        return true;
+                    }  else {
+                        return false;
+                    }
+                })
+                createTable(searchArr);
+                sumofChange (searchArr);
+            }
+        }
+    
+        let inputText = document.querySelector('.header-button') as HTMLInputElement;
+        inputText?.addEventListener('click', e => {
+            search();
+        })
+        
+    }
+    search();
+    return newArr;
+}
+
+function viewDisplay(view: string){
+    if (view === 'table') {
+        document.querySelector('.view-mode__table')?.classList.add ('view-active');
+        const productItem: NodeListOf<Element> = document.querySelectorAll('.product-item');
+        productItem.forEach (elem => {
+            if (elem instanceof HTMLDivElement) {
+                elem.style.flexDirection = 'column';
+                elem.style.width = '33.333%';
+            }
+            if (elem.childNodes[1].childNodes[2] instanceof HTMLDivElement) { 
+                elem.childNodes[1].childNodes[2].style.width = '100%';                  
+            }
+        })
+    }
+    if (view === 'list') {
+        document.querySelector('.view-mode__list')?.classList.add ('view-active');
+        const productItem: NodeListOf<Element> = document.querySelectorAll('.product-item');
+        productItem.forEach (elem => {
+            if (elem instanceof HTMLDivElement) {
+                elem.style.flexDirection = 'row';
+                elem.style.width = '100%';
+                if (elem.childNodes[0] instanceof HTMLImageElement) {
+                    elem.childNodes[0].style.maxWidth = '200px';
+                    elem.childNodes[0].style.marginRight = '20px';
+                }
+                if (elem.childNodes[1].childNodes[2] instanceof HTMLDivElement) { 
+                    elem.childNodes[1].childNodes[2].style.width = '50%';                  
+                }
+            }
+        })
+    }
+}
```

Comment:
Naming:
The function name viewDisplay could be improved to better describe what it does, for example updateProductView.

Typing:
The type of the view parameter is specified as string, but it would be better to use a enum type to restrict the possible values to either 'table' or 'list'.

DOM Manipulation:
The code heavily relies on accessing elements directly through the document object and using querySelector and querySelectorAll functions. This can lead to brittle code that breaks easily if the structure of the HTML changes. It might be a better approach to pass the relevant elements as arguments to the function instead.

Repetitiveness:
The two separate if statements are very similar in structure, with only small differences in the styles being applied. It might be more maintainable to abstract out the common parts of the code into a separate function and call it for both cases.

Error handling:
The code does not handle any errors that might occur, for example if the element is not found. The ?. operator makes sure that the code will not throw an error if the element is not found, but it will not have any effect on the styles applied. If it's necessary to handle missing elements, consider using a try-catch block.

Style:
The styles being applied to the elements are directly set on the elements themselves. It would be better to define CSS classes and toggle them on the elements instead, as this would make the code more readable and easier to maintain.

Accessibility:
The code does not consider accessibility requirements. For example, when changing the view mode to "list", it would be a good idea to add a role="list" attribute to the .product-item elements to make them semantically meaningful to assistive technologies.

Performance:
The code accesses the DOM elements and applies styles to them in a loop. This can be slow, especially if there are many elements on the page. It would be better to make use of CSS class toggles and minimize the amount of styles being applied.